homeassistant/helpers/entity_registry.py

```diff
@@ -102,7 +102,7 @@ def async_generate_entity_id(self, domain, suggested_object_id,
         Conflicts checked against registered and currently existing entities.
         """
         return ensure_unique_string(
-            '{}.{}'.format(domain, slugify(suggested_object_id)),
+            '{}.{}'.format(domain, slugify(suggested_object_id) or domain),
```

Comment:
This is not a good solution. That will generate so many duplicates.

Reply:
But problem only with non-english names. And duplicates will be resolved below.
I think, this is better solution than transliteration or generation random names.

Reply:
@balloob https://github.com/mozilla/unicode-slugify ?